### PR TITLE
html, xhr: Add tests for 'formdata' event

### PR DESCRIPTION
**Pull request by tkent-google**
This fixes a part of https://github.com/w3c/webcomponents/issues/187

HTML specification PR: https://github.com/whatwg/html/pull/4239
XHR specification PR: https://github.com/whatwg/xhr/pull/231

### CODE REVIEW

**Comment by tkent-google** [comment]
Taskcluster failures don't look related to this PR.

**Review by lukebjerring — COMMENTED** [review]
idlharness test changes LGTM

File: html/dom/interfaces.https.html [hunk 1]
```diff
@@ -33,11 +33,19 @@ <h1>HTML IDL tests</h1>
...
+  new FormData(form);
+  return formDataEvent;
+}
+
 const waitForLoad = new Promise(resolve => { addEventListener('load', resolve); })
 
 idl_test(
   ['html'],
-  ['SVG', 'cssom', 'touch-events', 'uievents', 'dom'],
+  ['SVG', 'cssom', 'touch-events', 'uievents', 'dom', 'xhr'],
```
Comment (annevk) [inline html/dom/interfaces.https.html]: Isn't the interface added to HTML?

**Review by annevk — COMMENTED** [review]
This looks fine to me, but per the HTML review we'll need to test the event constructor as well.

**Comment by annevk** [comment]
(I suspect you might be able to address the Taskcluster failure by rebasing on master. Apparently something changed over the holidays. Happy new year btw!)

File: html/dom/interfaces.https.html (hunk 1 above)
Comment (tkent-google) [reply html/dom/interfaces.https.html]: Yes.  However 'xhr' is necessary for FormData interface dependency.

**Review by annevk — APPROVED** [review]
I'll land this when the specification PR is ready.

**Comment by annevk** [comment]
I ran
```
git rebase master
git force --push
```
to see if I can get Taskcluster to pass.

**Comment by jgraham** [comment]
Taskcluster failures are for known flaky tests.

File: html/semantics/forms/form-submission-0/constructing-form-data-set.html [hunk 2]
```diff
@@ -46,4 +47,64 @@
...
+  let wasCancelable = true;
+  let form = populateForm();
+  document.addEventListener('formdata', e => {
+    didCallHandler = true;
+    wasBubbles = e.bubbles;
+    wasCancelable = e.cancelable;
+  });
+  new FormData(form);
+  assert_true(didCallHandler);
+  assert_true(wasBubbles);
```
Comment (marcoscaceres) [inline html/semantics/forms/form-submission-0/constructing-form-data-set.html]: @tkent-google shouldn't this be `assert_false`? Per the spec, it doesn't bubble. Or did I miss that somewhere?

File: html/semantics/forms/form-submission-0/constructing-form-data-set.html [hunk 3]
```diff
@@ -46,4 +47,64 @@
...
+test(() => {
+  let didCallHandler = false;
+  let form = populateForm();
+  let orphanRoot = document.createElement('div');
+  orphanRoot.appendChild(form);
+  orphanRoot.addEventListener('formdata', e => {
+    didCallHandler = true;
+  });
+  new FormData(form);
+  assert_true(didCallHandler);
```
Comment (marcoscaceres) [inline html/semantics/forms/form-submission-0/constructing-form-data-set.html]: Same as above, I guess... I'm reading this section:
https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#constructing-form-data-set:formdataevent

And the event fires as normal, and `bubbles = false` per the `EventInit` dict.

File: html/semantics/forms/form-submission-0/constructing-form-data-set.html (hunk 2 above)
Comment (tkent-google) [reply html/semantics/forms/form-submission-0/constructing-form-data-set.html]: Oops, it's a spec bug.  Please trust the test.
I'll make a specification PR.

Comment (tkent-google) [reply html/semantics/forms/form-submission-0/constructing-form-data-set.html]: PR: https://github.com/whatwg/html/pull/4422

**Pull request merged**